src/SFML/Graphics/Text.cpp

```diff
@@ -280,22 +293,29 @@ float Text::getOutlineThickness() const
 }
 
 
+////////////////////////////////////////////////////////////
+Text::LineAlignment Text::getLineAlignment() const
+{
+    return m_lineAlignment;
+}
+
+
 ////////////////////////////////////////////////////////////
 Vector2f Text::findCharacterPos(std::size_t index) const
 {
     // Adjust the index if it's out of range
     if (index > m_string.getSize())
         index = m_string.getSize();
 
+    // Calculate and update the line offsets
+    updateLineOffsets();
+
     // Precompute the variables needed by the algorithm
-    const bool  isBold          = m_style & Bold;
-    float       whitespaceWidth = m_font->getGlyph(U' ', m_characterSize, isBold).advance;
-    const float letterSpacing   = (whitespaceWidth / 3.f) * (m_letterSpacingFactor - 1.f);
-    whitespaceWidth += letterSpacing;
-    const float lineSpacing = m_font->getLineSpacing(m_characterSize) * m_lineSpacingFactor;
+    const bool isBold                                        = m_style & Bold;
+    const auto [whitespaceWidth, letterSpacing, lineSpacing] = getSpacing();
 
     // Compute the position
-    Vector2f      position;
+    Vector2f      position(m_lineOffsets[0], 0.f); // There will always be at least one line
```

Comment:
```suggestion
    assert(!m_lineOffsets.empty());
    Vector2f      position(m_lineOffsets[0], 0.f); // There will always be at least one line
```

Let's assert it then :)